### DISCUSSION TITLE
Zustand does not update when using maps

### DISCUSSION THREAD

**Discussion by Haberkamp** [pmndrs/zustand#1439] [Q&A] [answered]
Hello, I'm currently writing a jest test for a hook using a zustand store. Inside this hook I use an array and a map. Updating the array will be reactive but when I try to update the property using the map either by using the set method of the map or replacing the old map with a complete new map won't trigger a re-render. Does somebody know whats wrong with the code?

The test:
```javascript
const { result: hook } = renderHook(() => useCustomHook());

const foo = new Map();
foo.set('key', {
  type: 'add',
  data: 'value',
});

// ACT
await act(() => {
  useStore.setState({
    foo: foo,
  });
});

const result = hook.current.bla;

// ASSERT
// will throw an error because array is still empty, but it should not be
expect(result).toBeArrayOfSize(1)
```

// the hook:
```javascript
function useCustomHook() {
  // won't react to changes
  const foo = useStore(state => state.foo);

  // will react to changes
  const bar = useStore(state => state.bar);
}
```

**Comment by dai-shi** [comment, marked as answer]
https://github.com/pmndrs/zustand/blob/2b29d736841dc7b3fd7dec8cbfea50fee7295974/docs/guides/maps-and-sets-usage.md
Are you sure you are note mutating maps?

**Comment by Haberkamp** [comment]
Yes, I do and it works in the browser but the DOM inside the test file won't update

**Reply by Haberkamp** [reply]
Ahh, I found the issues. I have to hooks that are called very similar and I've used the wrong one. My bad... but thank you for you answer!
